pareto/models_extra/Integrate_desal/integrated_models/integrated_optimization_mvr.py

```diff
@@ -0,0 +1,274 @@
+                    for n in m.s_NTIN
+                    if n not in m.desalination_nodes
+                )
+                for t in m.s_T
+            )
+            * m.p_dt
+        )
+
+
+def integrated_model_build(network_data, treatment_dict={"R01_IN": 1}):
+    """
+    Inputs
+    -------
+    treatment_dict - A dictionary mapping treatment site to stages in the MEE-MVR unit
+    """
+    m = pyo.ConcreteModel()
+
+    # Network Model
+    m.m_network = build_network(network_data)
+
+    add_desalination_cons(m.m_network, treatment_dict)
+    manipulate_network_vars_and_cons(m.m_network)
+
+    # Solve network model
+    m.m_network.br_obj.deactivate()
+    m.m_network.obj.activate()
+    ipopt = pyo.SolverFactory("ipopt")
+
+    print("#### Initializing network ####")
+    ipopt.solve(m.m_network)
+
+    # Treatment models in each period
+    treatment_models = {}
+    global_evap_capex_index = []
+    for site in treatment_dict.keys():
+        N_evap = treatment_dict[site]
+        for i in m.m_network.s_T:
+            treatment_models[site, i] = make_mee_mvr_model(
+                N_evap=N_evap, inputs_variables=True
+            )
+
+        # Store desal site name and evaporator index for each in a
+        for n in range(N_evap):
+            global_evap_capex_index.append((site, n))
+    m.m_treatment = pyo.Reference(treatment_models)
+
+    # Initialize treatment models
+    print("#### Initializing desalination unit ####")
+    annual_fac = {}
+    for site in treatment_dict.keys():
+        for t in m.m_network.s_T:
+            m.m_treatment[site, t].flow_feed.fix(
+                pyo.value(sum(m.m_network.v_F[:, site, :, t]) * 0.0013)
+            )
+            m.m_treatment[site, t].salt_feed.fix(
+                pyo.value(m.m_network.v_Ctreatin[site, t])
+            )
+            ipopt.options["tol"] = 1e-6
+            res = ipopt.solve(m.m_treatment[site, i])
+            try:
+                pyo.assert_optimal_termination(res)
+            except:
+                import pdb
+
+                pdb.set_trace()
+            m.m_treatment[site, t].flow_feed.unfix()
+            m.m_treatment[site, t].salt_feed.unfix()
+
+            annual_fac[site] = pyo.value(m.m_treatment[site, t].annual_fac)
+
+    # Global capacity variables
+    # Indices for global evaporator capacity
+    m.global_evaporator_capex = pyo.Var(
+        global_evap_capex_index, domain=pyo.NonNegativeReals
+    )
+    m.global_preheater_capex = pyo.Var(
+        m.m_network.desalination_nodes, domain=pyo.NonNegativeReals
+    )
+    m.global_compressor_capex = pyo.Var(
+        m.m_network.desalination_nodes, domain=pyo.NonNegativeReals
+    )
+    m.CAPEX = pyo.Var(m.m_network.desalination_nodes, domain=pyo.NonNegativeReals)
+    m.OPEX = pyo.Var(
+        m.m_network.desalination_nodes, m.m_network.s_T, domain=pyo.NonNegativeReals
+    )
+
+    # ToDo: Global capacity variables initialization from treatment model solves
```

Comment:
Does this TODO need to be addressed? Please either address it or remove the comment.

Reply:
@Sakshi21299 can you please weigh in here? Alternatively, @JavalVyas2000 do you know the answer?

Reply:
I addressed this in 03d6074f05193d4e62ca4d95f9b254d817238d12 - resolving.